inc/saisystemport.h

```diff
@@ -0,0 +1,208 @@
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saisystemport.h
+ *
+ * @brief   This module defines SAI System Port interface
+ */
+
+#if !defined (__SAISYSTEMPORT_H_)
+#define __SAISYSTEMPORT_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAISYSTEMPORT SAI - System Port specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute data for #SAI_SYSTEM_PORT_ATTR_TYPE
+ */
+typedef enum _sai_system_port_type_t
+{
+    /** Local to line card */
+    SAI_SYSTEM_PORT_TYPE_LOCAL,
+
+    /** Remote to line card */
+    SAI_SYSTEM_PORT_TYPE_REMOTE,
+
+} sai_system_port_type_t;
+
+/**
+ * @brief Attribute Id in sai_set_system_port_attribute() and
+ * sai_get_system_port_attribute() calls
+ */
+typedef enum _sai_system_port_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_SYSTEM_PORT_ATTR_START,
+
+    /* READ-ONLY */
+
+    /**
+     * @brief System Port Type
+     *
+     * @type sai_system_port_type_t
+     * @flags READ_ONLY
+     */
+    SAI_SYSTEM_PORT_ATTR_TYPE = SAI_SYSTEM_PORT_ATTR_START,
+
+    /**
+     * @brief Number of Virtual output queues on port
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SYSTEM_PORT_ATTR_QOS_NUMBER_OF_VOQS,
+
+    /**
+     * @brief List of Virtual output Queues for the port
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_QUEUE
+     */
+    SAI_SYSTEM_PORT_ATTR_QOS_VOQ_LIST,
+
+    /**
+     * @brief Local port for the system port
+     *
+     * Only valid for system ports which are mapped to local ports.
+     *
+     * @type sai_object_id_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_PORT
+     */
+    SAI_SYSTEM_PORT_ATTR_PORT,
+
+    /* READ-WRITE */
+
+    /**
+     * @brief Speed in Mbps
+     *
+     * On get, returns the configured port speed.
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 10000
+     */
+    SAI_SYSTEM_PORT_ATTR_SPEED,
+
+    /**
+     * @brief Admin Mode
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_SYSTEM_PORT_ATTR_ADMIN_STATE,
+
```

Comment:
The intent of ADMIN_STATE is not clear for a system port. For local ports - does it mean that we can enable/disable a port using this attribute. For remote ports what does ADMIN_STATE do any way. If there is no clear intended usage - the suggestion is to remove this attribute. It could be added later if a need for such an attribute is discovered.

Reply:
I would like to keep this attribute. I feel we may find a need for it in the future. We can leave it up to the implementation as to whether to support it or not.